gcloud-java-dns/src/main/java/com/google/gcloud/spi/DnsRpc.java

```diff
@@ -52,5 +58,114 @@ Integer getInt(Map<Option, ?> options) {
     }
   }
 
-  //TODO(mderka) add supported operations. Created issue #594.
+  class Tuple<X, Y> {
+
+    private final X x;
+    private final Y y;
+
+    private Tuple(X x, Y y) {
+      this.x = x;
+      this.y = y;
+    }
+
+    public static <X, Y> Tuple<X, Y> of(X x, Y y) {
+      return new Tuple<>(x, y);
+    }
+
+    public X x() {
+      return x;
+    }
+
+    public Y y() {
+      return y;
+    }
+  }
+
+  /**
+   * Creates a new zone.
+   *
+   * @param zone a zone to be created
+   * @return Updated {@link ManagedZone} object
+   * @throws DnsException upon failure
+   */
+  ManagedZone create(ManagedZone zone) throws DnsException;
+
+  /**
+   * Retrieves and returns an existing zone.
+   *
+   * @param zoneName name of the zone to be returned
+   * @param options a map of options for the service call
+   * @return a zone or {@code null} if not found
+   * @throws DnsException upon failure
+   */
+  ManagedZone getZone(String zoneName, Map<Option, ?> options) throws DnsException;
+
+  /**
+   * Lists the zones that exist within the project.
+   *
+   * @param options a map of options for the service call
+   * @throws DnsException upon failure
+   */
+  Tuple<String, Iterable<ManagedZone>> listZones(Map<Option, ?> options) throws DnsException;
+
+  /**
+   * Deletes the zone identified by the name.
+   *
+   * @return {@code true} if the zone was deleted and {@code false} otherwise
+   * @throws DnsException upon failure
+   */
+  boolean deleteZone(String zoneName) throws DnsException;
+
+  /**
+   * Lists DNS records for a given zone.
+   *
+   * @param zoneName name of the zone to be listed
+   * @param options a map of options for the service call
+   * @throws DnsException upon failure or if zone not found
+   */
+  Tuple<String, Iterable<ResourceRecordSet>> listDnsRecords(String zoneName,
+      Map<Option, ?> options) throws DnsException;
+
+  /**
+   * Returns information about the current project.
+   * 
+   * @param options a map of options for the service call
+   * @return up-to-date project information
+   * @throws DnsException upon failure
+   */
+  Project getProject(Map<Option, ?> options) throws DnsException;
```

Comment:
I think this should never happen in the first place. However, this is kind of the scenario when the parent container does not exist, is it not? Furthermore, it feels consistent with other parts of the library. Say that you want to fetch a bucket from non-existent project using Storage, you will get an exception as well, right?

The service call itself gives us 400 error.